lib/chef/provider/package/freebsd/port.rb

```diff
@@ -26,32 +26,44 @@ class Port < Base
           include PortsHelper
 
           def install_package(name, version)
-            shell_out_with_timeout!("make -DBATCH install clean", :timeout => 1800, :env => nil, :cwd => port_dir).status
+            [name].flatten.each do |name|
+              shell_out_with_timeout!("make -DBATCH install clean", :timeout => 1800, :env => nil, :cwd => port_dir(name)).status
+            end
           end
 
           def remove_package(name, version)
-            shell_out_with_timeout!("make deinstall", :timeout => 300, :env => nil, :cwd => port_dir).status
+            [name].flatten.each do |name|
+              shell_out_with_timeout!("make deinstall", :timeout => 300, :env => nil, :cwd => port_dir(name)).status
+            end
           end
 
-          def current_installed_version
-            pkg_info = if @new_resource.supports_pkgng?
-                         shell_out_with_timeout!("pkg info \"#{@new_resource.package_name}\"", :env => nil, :returns => [0, 70])
-                       else
-                         shell_out_with_timeout!("pkg_info -E \"#{@new_resource.package_name}*\"", :env => nil, :returns => [0, 1])
-                       end
-            pkg_info.stdout[/^#{Regexp.escape(@new_resource.package_name)}-(.+)/, 1]
+          # The name of the package (without the version number) as understood by pkg_add and pkg_info.
+          def package_name
+            [@new_resource.package_name].flatten.map do |name|
```

Comment:
multipackage supporting providers need to declare use_multipackage_api and then they'll always be passed arrays instead of scalars.  you need to also use some other conventions like always using package_name_array from the superclass.

i would suggest dropping the multipackage support out of this PR, its going to complicate it all and probably cause a lot of frustration.